src/mongocxx/pool.hpp

```diff
@@ -55,6 +55,9 @@ class MONGOCXX_API pool {
     ///  A MongoDB URI representing the connection parameters
     /// @param ssl_options
     ///  Optional SSL options to use when connecting to the MongoDB deployment.
+    ///
+    /// @throws mongocxx::exception if ssl_options are invalid
```

Comment:
Taking a look at the logic for when SSL is not enabled, what do you think instead about `@throws mongocxx::exception if SSL is enabled and ssl_options are invalid, or if SSL is not enabled and ssl_options is engaged`?